pkg/apis/pipeline/v1beta1/taskrun_types.go

```diff
@@ -88,6 +88,9 @@ const (
 	// TaskRunSpecStatusCancelled indicates that the user wants to cancel the task,
 	// if not already cancelled or terminated
 	TaskRunSpecStatusCancelled = "TaskRunCancelled"
+	// TaskRunSpecStatusTimedOut indicates that the PipelineRun owning this task wants to mark it as timed out,
+	// if not already cancelled or terminated
+	TaskRunSpecStatusTimedOut = "TaskRunTimedOut"
```

Comment:
I'm not sure I understand why this is going in the spec. SpecStatusCancelled is for letting a user cancel the TaskRun, but I'm not sure we'd want to be able to set the spec to TimedOut

Reply:
@jerop's idea was to leverage the same approach we use when cancelling a `PipelineRun` to get its child `TaskRun`s and `Run`s to cancel as well, which is to set `spec.status` to`TaskRunSpecStatusCancelled`. If there's another way to signal the `TaskRun` that it should stop itself and report as timed out, I'm more than happy to see it. =)

Reply:
could we reuse the timeout strategy we use for runs? https://github.com/tektoncd/pipeline/blob/d3b8017e652115ac102a5d10fc455240c1331a49/pkg/reconciler/pipelinerun/pipelinerun.go#L623-L648

Reply:
I tend to agree with @abayer there

Reply:
I would argue that if it's the `PipelineRun` that times out, the `TaskRun` itself did not timeout, but it was cancelled because of some timeout logic that the `TaskRun` itself is not aware of. 
I think using cancellation might actually help users identity where the timeout occurred. We could add a place for the canceller to give an explanation or reason for the cancellation.
Using cancellation might be even more helpful when doing pipeline in pipeline, where the chain of cancellations may go deeper.

Reply:
👍

Reply:
I think a user shouldn't need to provide a cancellation reason, and the PR controller can provide the reason for cancelling the taskrun, so I'm not sure we need to change `pipelineRun.spec.status`, however we might need to add a field to `taskrun.spec.status`

Reply:
Oops, yeah, I meant `taskRun.spec.status`. Also I forgot to delete this in my latest commit. heh.

Reply:
I agree a user shouldn't need to provide a cancellation reason, but they might want to - it could be an interesting possibility to give our users for auditing. That said, the same could be achieved a bit more loosely with annotations.